posthog/api/services/flag_definitions_cache.py

```diff
@@ -0,0 +1,229 @@
+"""
+Cache management for flag definitions used in local evaluation.
+
+This module provides a centralized way to manage caching of flag definitions,
+which includes feature flags, group type mappings, and cohorts.
+"""
+
+import logging
+import os
+from typing import Any, Optional
+
+from django.core.cache import cache
+
+
+class FlagDefinitionsCache:
+    """Manages caching for flag definitions used in local evaluation."""
+
+    # Cache configuration
+    CACHE_VERSION = "v1"
+    CACHE_TTL = int(os.getenv("FLAG_DEFINITIONS_CACHE_TTL", 600))  # Default: 10 minutes
```

Comment:
Did you want to do an hour still?

Reply:
Yeah, I think I lost that work when I had the Git mishap. I'll fix.